business_register/views/company_views.py

```diff
@@ -5,14 +5,20 @@
 from business_register.models.company_models import Company, HistoricalCompany
 from business_register.serializers.company_serializers import CompanySerializer, HistoricalCompanySerializer
 from data_ocean.views import CachedViewMixin
+from rest_framework.filters import SearchFilter
 
 
 class CompanyView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
     permission_classes = [AllowAny]
-    queryset = Company.objects.all()
+    queryset = Company.objects.select_related('parent', 'company_type', 'status', 'authority',
+                                              'bylaw') \
+        .prefetch_related('founders', 'predecessors', 'assignees', 'signers', 'kveds',
+                          'bancruptcy_readjustment', 'company_detail', 'exchange_data',
+                          'termination_started').all()
```

Comment:
One more little change))
```suggestion
    queryset = Company.objects.select_related(
        'parent', 'company_type', 'status', 'authority', 'bylaw',
    ).prefetch_related(
        'founders', 'predecessors', 'assignees', 'signers', 'kveds', 'termination_started',
        'bancruptcy_readjustment', 'company_detail', 'exchange_data',
    ).all()
```